savant/utils/zeromq.py

```diff
@@ -371,3 +374,17 @@ def set_ipc_socket_permissions(socket: str, permission: int = 0o777):
     if parsed.scheme == 'ipc':
         logger.debug('Setting socket permissions to %o (%s).', permission, socket)
         os.chmod(parsed.path, permission)
+
+
+def create_ipc_socket_dirs(socket: str):
```

Comment:
Python's `os` package deals with this difference by introducing 2 functions: `os.mkdir()` and `os.makedirs()`. Since this function relies on `os.makedirs()` I think it makes sense to follow this naming choice.